src/Application/Services/ItemService.cs

```diff
@@ -39,86 +46,97 @@ public async Task<Item> Get(string key)
+        }
+
+    }
+
+    public async Task<string> Create(ItemCreate itemDto)
+    {
+        if (itemDto.ExpirationPeriod < MaxItemExPeriod)
+        {
+            if (await _itemRepository.Get(itemDto.Key) is null)
             {
-                Key = itemDto.Key,
-                Value = JsonSerializer.Serialize(itemDto.Value),
-                ExpirationPeriod = ExPeriod,
-                ExpirationDate = DateTime.Now.AddSeconds(ExPeriod)
-            };
+                int exPeriod = itemDto.ExpirationPeriod is 0 ? DefaultExPeriod : itemDto.ExpirationPeriod.Value;
+
+                ItemEntity itemEntity = new()
+                {
+                    Key = itemDto.Key,
+                    Value = JsonSerializer.Serialize(itemDto.Value),
+                    ExpirationPeriod = exPeriod,
+                    ExpirationDate = DateTime.Now.AddSeconds(exPeriod)
+                };
+
+                await _itemRepository.Create(itemEntity);
 
-            await _itemRepository.Create(itemEntity);
+                return "Item created";
+            }
 
-            return "Item created";
+            throw new FoundException("Item with same key already exists.");
         }
+
+        throw new ArgumentException($"Invalid ExpirationPeriod field");
     }
 
     public async Task<string> Update(ItemCreate itemDto)
     {
-        if (await _itemRepository.Get(itemDto.Key) is null)
+        if (itemDto.ExpirationPeriod < MaxItemExPeriod)
         {
-            int ExPeriod = itemDto.ExpirationPeriod ?? 0;
-
-            ItemEntity itemEntity = new()
+            if (await _itemRepository.Get(itemDto.Key) is null)
             {
-                Key = itemDto.Key,
-                Value = JsonSerializer.Serialize(itemDto.Value),
-                ExpirationPeriod = ExPeriod,
-                ExpirationDate = DateTime.Now.AddSeconds(ExPeriod)
-            };
+                int exPeriod = itemDto.ExpirationPeriod is 0 ? DefaultExPeriod : itemDto.ExpirationPeriod.Value;
 
-            await _itemRepository.Create(itemEntity);
+                ItemEntity itemEntity = new()
+                {
+                    Key = itemDto.Key,
+                    Value = JsonSerializer.Serialize(itemDto.Value),
+                    ExpirationPeriod = exPeriod,
+                    ExpirationDate = DateTime.Now.AddSeconds(exPeriod)
+                };
 
-            return "Item created";
-        }
-        else
-        {
-            int ExPeriod = itemDto.ExpirationPeriod ?? 0;
+                await _itemRepository.Create(itemEntity);
 
-            ItemEntity itemEntity = new()
+                return "Item created";
+            }
+            else
             {
-                Key = itemDto.Key,
-                Value = JsonSerializer.Serialize(itemDto.Value),
-                ExpirationPeriod = ExPeriod
-            };
+                int exPeriod = itemDto.ExpirationPeriod is 0 ? DefaultExPeriod : itemDto.ExpirationPeriod.Value;
```

Comment:
incorrect code, should be "If expiration period is not defined - default value should be used from config"